patches/minecraft/net/minecraft/world/server/ServerWorld.java.patch

```diff
@@ -315,7 +315,37 @@
+    }
+ 
+    public <T extends IParticleData> int func_195598_a(T p_195598_1_, double p_195598_2_, double p_195598_4_, double p_195598_6_, int p_195598_8_, double p_195598_9_, double p_195598_11_, double p_195598_13_, double p_195598_15_) {
+-      SSpawnParticlePacket sspawnparticlepacket = new SSpawnParticlePacket(p_195598_1_, false, (float)p_195598_2_, (float)p_195598_4_, (float)p_195598_6_, (float)p_195598_9_, (float)p_195598_11_, (float)p_195598_13_, (float)p_195598_15_, p_195598_8_);
++      net.minecraftforge.event.ParticleWorldHandleEvent event = new net.minecraftforge.event.ParticleWorldHandleEvent(p_195598_1_, false, p_195598_8_, p_195598_15_, p_195598_2_, p_195598_4_, p_195598_6_, p_195598_9_, p_195598_11_, p_195598_13_);
++      if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) return 0;
++      SSpawnParticlePacket sspawnparticlepacket = new SSpawnParticlePacket(event.getParticle(), event.isLongDistance(), (float)event.getXCoord(), (float)event.getYCoord(), (float)event.getZCoord(), (float)event.getXOffset(), (float)event.getYOffset(), (float)event.getZOffset(), (float)event.getParticleSpeed(), event.getParticleCount());
+       int i = 0;
+ 
+       for(int j = 0; j < this.field_217491_A.size(); ++j) {
+          ServerPlayerEntity serverplayerentity = this.field_217491_A.get(j);
+-         if (this.func_195601_a(serverplayerentity, false, p_195598_2_, p_195598_4_, p_195598_6_, sspawnparticlepacket)) {
++         if (this.func_195601_a(serverplayerentity, event.isLongDistance(), event.getXCoord(), event.getYCoord(), event.getZCoord(), sspawnparticlepacket)) {
+             ++i;
+          }
+       }
+@@ -1113,8 +1161,10 @@
+    }
+ 
+    public <T extends IParticleData> boolean func_195600_a(ServerPlayerEntity p_195600_1_, T p_195600_2_, boolean p_195600_3_, double p_195600_4_, double p_195600_6_, double p_195600_8_, int p_195600_10_, double p_195600_11_, double p_195600_13_, double p_195600_15_, double p_195600_17_) {
+-      IPacket<?> ipacket = new SSpawnParticlePacket(p_195600_2_, p_195600_3_, (float)p_195600_4_, (float)p_195600_6_, (float)p_195600_8_, (float)p_195600_11_, (float)p_195600_13_, (float)p_195600_15_, (float)p_195600_17_, p_195600_10_);
+-      return this.func_195601_a(p_195600_1_, p_195600_3_, p_195600_4_, p_195600_6_, p_195600_8_, ipacket);
++      net.minecraftforge.event.ParticleWorldHandleEvent event = new net.minecraftforge.event.ParticleWorldHandleEvent(p_195600_2_, false, p_195600_10_, p_195600_17_, p_195600_4_, p_195600_6_, p_195600_8_, p_195600_11_, p_195600_13_, p_195600_15_);
++      if (net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(event)) return false;
++      IPacket<?> ipacket = new SSpawnParticlePacket(event.getParticle(), event.isLongDistance(), (float)event.getXCoord(), (float)event.getYCoord(), (float)event.getZCoord(), (float)event.getXOffset(), (float)event.getYOffset(), (float)event.getZOffset(), (float)event.getParticleSpeed(), event.getParticleCount());
```

Comment:
Could put this helper function inside the event? it could make it more readable

Reply:
You mean creating the SSpawnParticlePacket inside the event?

Reply:
yeah just to be able to read it better maybe and less clutter? 
then you can just do event.getPacket(); instead of that huge line